cmd/mdatagen/internal/metadata.go

```diff
@@ -267,6 +267,58 @@ type Attribute struct {
 	Warnings Warnings `mapstructure:"warnings"`
 }
 
+// Name returns actual name of the attribute that is set on the metric after applying NameOverride.
+func (a ResourceAttribute) Name() AttributeName {
+	if a.NameOverride != "" {
+		return AttributeName(a.NameOverride)
+	}
+	return a.FullName
+}
+
+func (a ResourceAttribute) TestValue() string {
+	if a.Enum != nil {
+		return fmt.Sprintf(`"%s"`, a.Enum[0])
+	}
+	switch a.Type.ValueType {
+	case pcommon.ValueTypeEmpty:
+		return ""
+	case pcommon.ValueTypeStr:
+		return fmt.Sprintf(`"%s-val"`, a.FullName)
+	case pcommon.ValueTypeInt:
+		return strconv.Itoa(len(a.FullName))
+	case pcommon.ValueTypeDouble:
+		return fmt.Sprintf("%f", 0.1+float64(len(a.FullName)))
+	case pcommon.ValueTypeBool:
+		return strconv.FormatBool(len(a.FullName)%2 == 0)
+	case pcommon.ValueTypeMap:
+		return fmt.Sprintf(`map[string]any{"key1": "%s-val1", "key2": "%s-val2"}`, a.FullName, a.FullName)
+	case pcommon.ValueTypeSlice:
+		return fmt.Sprintf(`[]any{"%s-item1", "%s-item2"}`, a.FullName, a.FullName)
+	case pcommon.ValueTypeBytes:
+		return fmt.Sprintf(`[]byte("%s-val")`, a.FullName)
+	}
+	return ""
+}
+
+type Attribute struct {
```

Comment:
Can we embed this struct in `ResourceAttribute` with `mapstructure:",squash"` to avoid repeating the fields and methods? That

Reply:
sure thing